.gitignore

```diff
@@ -7,3 +7,5 @@ build/
 build-*/
 wayland-*-protocol.*
 wlr-example.ini
+compile_commands.json
+.cache/
```

Comment:
Where are these files coming from? Can't we somehow put these in the build dir?

Reply:
Looks like files from a language server like [ccls](https://github.com/MaskRay/ccls) - the compile_commands.json can be created by ninja and marks the project root for ccls as well as tells it how to compile stuff. So afaik it can't easily be moved to build/. Still, those should be excluded locally I think and not pollute the .gitignore file.

Reply:
Yeah `.cache` comes from the clangd language server in my case and `compile_commands.json` is simply linked from the build dir. I you prefer not cluttering the gitignore file I can remove it from this PR but the workflow isn't too uncommon (and removes the need for the `.clang_complete` file).